src/mono/CMakeLists.txt

```diff
@@ -505,9 +505,11 @@ if(GCC)
   set(WERROR "-Werror=return-type")
   set(WERROR_C "-Werror=implicit-function-declaration")
 
-  check_c_compiler_flag("-Werror=incompatible-pointer-types" WERROR_INCOMPATIBLE_POINTER_TYPES)
-  if(WERROR_INCOMPATIBLE_POINTER_TYPES)
-    set(WERROR_C "${WERROR_C} -Werror=incompatible-pointer-types")
+  if(NOT HOST_WASI)
+    check_c_compiler_flag("-Werror=incompatible-pointer-types" WERROR_INCOMPATIBLE_POINTER_TYPES)
```

Comment:
Why is this needed ?

Reply:
When building under WASI it's being stricter about discarding `const` qualifiers. Existing code elsewhere (e.g., mono-rand.c:315), which I'm trying not to change in non-WASI builds, results in an error about that.

I've updated the config now so it's more precise about adding `no-incompatible-pointer-types-discards-qualifiers` for the WASI build rather than skipping `incompatible-pointer-types` entirely, so that should be more robust at least.